bootstrap

```diff
@@ -114,17 +118,26 @@ if [ -z "$NO_CLOBBER" ] && \
   [ ! -e "$VENV/bin/pip" -o ! -e "$VENV/bin/$PYTHON" ] || \
   [ "${PYTHON_VER_OUT}" != "$($VENV/bin/$PYTHON --version)" ] \
   ; then
+   echo "Deleting existing virtual environment"
     rm -rf virtualenv
 fi
 
 if [ -z "$NO_CLOBBER" ] || [ ! -e $VENV ]; then
+    echo "Creating new venv at $VENV"
     $PYTHON_BIN -m venv $VENV
 fi
 
+PY_MAJOR=$($VENV/bin/python -c "import sys; print(sys.version_info[0])")
+PY_MINOR=$($VENV/bin/python -c "import sys; print(sys.version_info[1])")
+
+# Downgrade pip for older python versions
+if [ "$PY_MAJOR" -eq 3 ] && [ "$PY_MINOR" -lt 6 ]; then
```

Comment:
We require python 3.10 because we use features introduced in that version: https://github.com/ceph/teuthology/blob/main/setup.cfg#L29